src/lib/jsonParseV2/parseSubscription.cpp

```diff
@@ -303,11 +308,14 @@ static std::string parseEntitiesVector(ConnectionInfo* ciP, std::vector<EntID>*
       {
         return badInput(ciP, idOpt.error);
       }
-      id = idOpt.value;
-
-      if (id.empty())
+      else if (idOpt.given)
       {
-        idPattern = ".*";
+        if (forbiddenIdCharsV2(idOpt.value.c_str()))
+        {
+          return badInput(ciP, "forbidden characters in subject entities element id");
+        }
+
+        id= idOpt.value;
       }
```

Comment:
Now, what about id: "" and idPattern: "" ?
Perhaps this is taken care of by the forbidden char checks ... ?